### PR TITLE
kie-issues#243: Fill the "description" of every env var on `kie-tools` build and save the report somewhere (1/2)

### PR DESCRIPTION
**Pull request by tiagobento**
Part of https://github.com/apache/incubator-kie-issues/issues/243

Still missing the report. Will come in a future PR.

Also on this PR:
- Removing `podman` references and commands
- Renaming `buildTags` to `buildTag`. Only one tag version of images is supported during the build. Retagging needs to be done separately. 
- Improving `build-env` with `--help` and `--generate-empty-env-index-js`

### CODE REVIEW

File: packages/chrome-extension-serverless-workflow-editor/env/index.js [hunk 1]
```diff
@@ -23,19 +23,19 @@ module.exports = composeEnv([require("@kie-tools/root-env/env")], {
...
     },
     SWF_CHROME_EXTENSION__manifestFile: {
       default: "manifest.dev.json",
-      description: "",
+      description: "Chrome Extension manifest file path relative to this package's root dir.",
     },
     SWF_CHROME_EXTENSION__e2eTestingToken: {
       default: "",
-      description: "",
+      description: "", // TODO: Fill this
```
Comment (Josephblt) [inline packages/chrome-extension-serverless-workflow-editor/env/index.js]: Did you leave this TODO intentionally? If so, please disregard it.

File: packages/webpack-base/env/index.js [hunk 2]
```diff
@@ -23,19 +23,23 @@ module.exports = composeEnv([], {
   vars: varsWithName({
     WEBPACK__minimize: {
       default: undefined,
-      description: "",
+      description:
+        "Whether or not `webpack` should minimize its bundled files. Can be `true` or `false`. Works for all packages on invokations of `webpack` during a `kie-tools` build.",
```
Comment (Josephblt) [inline packages/webpack-base/env/index.js]: I think I found a typo here. Invokation should be Invocation, perhaps? If it is in fact called invocation, please disregard.

File: packages/webpack-base/env/index.js [hunk 3]
```diff
@@ -23,19 +23,23 @@ module.exports = composeEnv([], {
...
     },
     WEBPACK__tsLoaderTranspileOnly: {
       default: undefined,
-      description: "",
+      description:
+        "Whether or not `webpack` should skip type-checking TypeScript files. Can be `true` or `false`. Works for all packages on invokations of `webpack` during a `kie-tools` build.",
```
Comment (Josephblt) [inline packages/webpack-base/env/index.js]: Same as above, Invokation typo.

File: packages/webpack-base/env/index.js [hunk 4]
```diff
@@ -23,19 +23,23 @@ module.exports = composeEnv([], {
...
     },
     WEBPACK__sourceMaps: {
       default: undefined,
-      description: "",
+      description:
+        "Whether or not `webpack` should include source maps for its bundled files. Can be `true` or `false`. Works for all packages on invokations of `webpack` during a `kie-tools` build.",
```
Comment (Josephblt) [inline packages/webpack-base/env/index.js]: Same as above, Invokation typo.

File: packages/webpack-base/env/index.js [hunk 5]
```diff
@@ -23,19 +23,23 @@ module.exports = composeEnv([], {
...
     },
     WEBPACK__mode: {
       default: undefined,
-      description: "",
+      description:
+        "Can be `development` or `production`. Works for all packages on invokations of `webpack` during a `kie-tools` build.",
```
Comment (Josephblt) [inline packages/webpack-base/env/index.js]: Same as above, Invokations typo.

File: packages/dev-deployment-upload-service/env/index.js [hunk 6]
```diff
@@ -35,15 +35,15 @@ module.exports = composeEnv([rootEnv], {
     },
     DEV_DEPLOYMENT_UPLOAD_SERVICE__devFileServerPort: {
       default: 2340,
-      description: "",
+      description: "", // TODO: Fill
```
Comment (Josephblt) [inline packages/dev-deployment-upload-service/env/index.js]: Perhaps you forgot this here? If not please disregard.

File: packages/dev-deployment-upload-service/env/index.js [hunk 7]
```diff
@@ -35,15 +35,15 @@ module.exports = composeEnv([rootEnv], {
...
     },
     DEV_DEPLOYMENT_UPLOAD_SERVICE__devBuildTimeInstallPort: {
       default: 2341,
-      description: "",
+      description: "", // TODO: Fill
```
Comment (Josephblt) [inline packages/dev-deployment-upload-service/env/index.js]: Same as above.

File: packages/dev-deployment-upload-service/env/index.js [hunk 8]
```diff
@@ -35,15 +35,15 @@ module.exports = composeEnv([rootEnv], {
...
     },
     DEV_DEPLOYMENT_UPLOAD_SERVICE__devRunTimeInstallPort: {
       default: 2342,
-      description: "",
+      description: "", // TODO: Fill
```
Comment (Josephblt) [inline packages/dev-deployment-upload-service/env/index.js]: Same as above.

**Review by Josephblt — CHANGES_REQUESTED** [review]
Hello @tiagobento. I left some questions regarding some TODO's and a small typo.

**Comment by tiagobento** [comment]
@Josephblt Thanks! Please check https://github.com/apache/incubator-kie-tools/pull/2421/commits/428da0a4dd8e3f1d17f278b5792619814d0dab17

**Review by Josephblt — APPROVED** [review]

**Review by thiagoelg — APPROVED** [review]

**Pull request merged**